force-app/main/default/classes/GeocodingService.cls

```diff
@@ -0,0 +1,75 @@
+public with sharing class GeocodingService {
+    @InvocableMethod(label='Geocode address')
+    public static List<Coordinates> geocodeAddresses(
+        List<GeocodingAddress> addresses
+    ) {
+        List<Coordinates> computedCoordinates = new List<Coordinates>();
+
+        String baseUrl = 'https://nominatim.openstreetmap.org/search?format=json';
+
+        for (GeocodingAddress address : addresses) {
+            String url = baseUrl;
+            url += (String.isNotBlank(address.street))
+                ? '&street=' + address.street
+                : '';
+            url += (String.isNotBlank(address.city))
+                ? '&city=' + address.city
+                : '';
+            url += (String.isNotBlank(address.state))
+                ? '&state=' + address.state
+                : '';
+            url += (String.isNotBlank(address.country))
+                ? '&country=' + address.country
+                : '';
+            url += (String.isNotBlank(address.postalcode))
+                ? '&postalcode=' + address.postalcode
+                : '';
+
+            Boolean found = false;
```

Comment:
I would implement that in a simpler way:
- Instead of using a boolean, declare `Coordinates  coords = new Coordinates();`
- Overwrite the value of `coords` if you have retrieved it from the service
- Always do: `computedCoordinates.add(coords);`